python-backend/app/api/mines/variances/resources/variance.py

```diff
@@ -0,0 +1,238 @@
+            'filename': document_name
+        }
+        document_manager_URL = f'{current_app.config["DOCUMENT_MANAGER_URL"]}/document-manager'
+
+        resp = requests.post(
+            url=document_manager_URL,
+            headers={key: value
+                     for (key, value) in request.headers if key != 'Host'},
+            data=data,
+            cookies=request.cookies,
+        )
+
+        response = Response(str(resp.content), resp.status_code, resp.raw.headers.items())
+        return response
+
+    @api.doc(
+        description=
+        'Associate an uploaded file with a variance.',
+        params={
+            'mine_guid': 'guid for the mine with which the variance is associated',
+            'variance_id': 'ID for the variance to which the document should be associated'
+        }
+    )
+    @api.marshal_with(variance_model, code=200)
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def put(self, mine_guid, variance_id):
+        variance = Variance.find_by_variance_id(variance_id)
+
+        data = self.parser.parse_args()
+        document_name = data.get('document_name')
+        document_manager_guid = data.get('document_manager_guid')
+
+        if not document_name:
+            raise BadRequest('Missing document_name')
+
+        # Register new file upload
+        mine_doc = MineDocument(
+            mine_guid=mine_guid,
+            document_manager_guid=document_manager_guid,
+            document_name=document_name)
+
+        if not mine_doc:
+            raise BadRequest('Unable to register uploaded file as document')
+
+        variance.documents.append(mine_doc)
+        db.session.commit()
+        return variance
+
+
+    def _parse_request_metadata(self):
+        request_metadata = request.headers.get("Upload-Metadata")
+        metadata = {}
+        if not request_metadata:
+            return metadata
+
+        for key_value in request_metadata.split(","):
+            (key, value) = key_value.split(" ")
+            metadata[key] = base64.b64decode(value).decode("utf-8")
+
+        return metadata
+
+
+class VarianceUploadedDocumentsResource(Resource, UserMixin, ErrorMixin):
+    @api.doc(
+        description=
+        'Delete a document from a variance.'
+    )
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def delete(self, mine_guid, variance_id, mine_document_guid):
+        variance = Variance.find_by_variance_id(variance_id)
+        mine_document = MineDocument.find_by_mine_document_guid(mine_document_guid)
+
+        if variance is None:
+            raise NotFound('Variance not found')
+        if mine_document is None:
+            raise NotFound('Mine document not found')
+
+        variance.documents.remove(mine_document)
+        variance.save()
+
+        # FIXME: This response is inconsistent with the others
+        return {'status': 204, 'message': 'The document was removed succesfully'}
```

Comment:
can this be
```return ('',204)``` 

204 means 'success but no response body'

Reply:
I tried that before based on your comment on the original PR, and it just blew up. I'll revisit it now

Reply:
Odd, we are doing that elsewhere.... wonder if flask-restplus wants something else? also try ```return '', 204```